test/validation/api/crypto/odp_crypto_test_inp.c

```diff
@@ -444,9 +477,9 @@ static void alg_test_execute(const alg_test_param_t *param)
 	uint8_t *cipher_iv = param->override_iv ? ref->cipher_iv : NULL;
 	uint8_t *auth_iv   = param->override_iv ? ref->auth_iv : NULL;
 
-	cipher_range.offset = 0;
+	cipher_range.offset = param->header_len;
```

Comment:
For bit mode algorithms, as per the current spec, cipher_range.offset and auth_range.offset should be in bits not bytes.

Apologies for the delayed response. We tried verifying this on our platform and ran into failures with bit mode algorithms and had been debugging the same.

Reply:
Thanks for testing and investigating this. The interesting thing is that validation tests with linux-gen pass now but with the fix they fail. It seems the linux-gen implementation of bit mode crypto and the implementation of EIA2 CMAC are buggy in that they assume the offset is in bytes.

Reply:
I made a new PR (1380) to fix both the validation test and linux-gen implementation. Please take a look if it fixes the validation problem with your ODP.

Reply:
Thank you @JannePeltonen for the new PR to fix this. Will verify with the new PR and update you there.